src/scenic/simulators/webots/model.scenic

```diff
@@ -119,6 +153,89 @@ class Ground(WebotsObject):
             y += dy
         return tuple(heights)
 
+    @staticmethod
+    @distributionFunction
+    def shapeFromHeights(heights, width, length, gridSizeX, gridSizeY, baseThickness):
+        heights = [row[::-1] for row in heights[::-1]]
+
+        triangles = []
+
+        ## Vertices
+        dx, dy = width / (gridSizeX - 1), length / (gridSizeY - 1)
+        base_x, base_y = -width / 2, length / 2
+
+        heightmap_vertices = tuple(tuple((base_x+ix*dx, base_y-iy*dy, heights[iy][ix]) for ix in range(gridSizeX)) for iy in range(gridSizeY))
+
+        raw_vertices = [v for row in heightmap_vertices for v in row]
+        top_vertices = list(dict.fromkeys(raw_vertices))
+        vertices =  top_vertices + \
+                    [( width/2,-length/2,-baseThickness),
+                     (-width/2,-length/2,-baseThickness),
+                     (-width/2, length/2,-baseThickness),
+                     ( width/2, length/2,-baseThickness),]
+        vertex_index_map = {vertices[i]:i for i in range(len(vertices))}
+
+        # Create top surface
+        for ix in range(gridSizeX-1):
+            for iy in range(gridSizeY-1):
+                # Calculate an interpolated middle point
+                tl_x, tl_y, _ = heightmap_vertices[ix][iy]
+                bl_x, bl_y, _ = heightmap_vertices[ix+1][iy+1]
+                mean_height = (heightmap_vertices[ix][iy][2] + heightmap_vertices[ix+1][iy][2] + heightmap_vertices[ix][iy+1][2] + heightmap_vertices[ix+1][iy+1][2])/4
+                interpolated_point = ((tl_x + bl_x)/2, (tl_y + bl_y)/2, mean_height)
+
+                # Left
+                triangles += [(heightmap_vertices[ix][iy], interpolated_point, heightmap_vertices[ix][iy+1])]
+                # Top
+                triangles += [(heightmap_vertices[ix][iy], heightmap_vertices[ix+1][iy], interpolated_point)]
+                # Right
+                triangles += [(heightmap_vertices[ix+1][iy], heightmap_vertices[ix+1][iy+1], interpolated_point)]
+                # Bottom
+                triangles += [(heightmap_vertices[ix][iy+1], interpolated_point, heightmap_vertices[ix+1][iy+1])]
```

Comment:
Building 4 lists of length 1 is confusing to me: I would either call `append` 4 times or (better) `extend` once, i.e.:
```python
triangles.extend([
    (v1, v2, v3),  # Left
    ...
])